### DISCUSSION TITLE
Is it safe to run multiple beets (database altering) sessions at once?

### DISCUSSION THREAD

**Discussion by BrianPuccio** [beetbox/beets#4203] [Q&A] [answered]
I found [this post on the old discourse forums](https://discourse.beets.io/t/is-it-save-to-start-many-db-modifying-instances-of-beets/1125), but I was wondering if anything had changed.

My use case is similar: while one (very large) import is churning away in the background, can I work on several more imports in another terminal with no ill effect? FWIW, the artists won't overlap so I don't run into a two imports trying to create the same artist at the same time.

**Comment by sampsyo** [comment, marked as answer]
I think my advice remains more or less the same. The actual data integrity of the database should be fine because of SQLite’s transactional semantics, and in my experience, few weird things happen when using the database in parallel. (And yeah, all that reassurance is off if the database is on an NFS volume or whatever.) I can definitely imagine weird corner cases cropping up with replacing or re-importing the same album in different processes, but none of that is really relevant if you think you'll be touching disjoint sets of albums in the parallel processes.

Good luck!
